### PR TITLE
Remove all QMC_CUDA dependencies on testsets

### PR DESCRIPTION
**Pull request by Hyeondeok-Shin**
## Proposed changes

Since legacy CUDA is removed on #4431, QMC_CUDA dependencies on testsets are no longer needed.

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Other (please describe): Clean up testsets

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
Intel Xeon

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- No. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

**Review by prckent — DISMISSED** [review]
Confirmed that QMC_CUDA is gone from tests/*

Approving, of course subject to CI

**Comment by prckent** [comment]
Test this please

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -40,11 +40,7 @@ else()
   set(SUCCESS_STATUS_OFFLOAD TRUE)
 endif()
 
-if(QMC_CUDA)
-  set(SUCCESS_STATUS_CUDA FALSE)
-else()
-  set(SUCCESS_STATUS_CUDA TRUE)
-endif()
+set(SUCCESS_STATUS_CUDA TRUE)
```
Comment (ye-luo) [inline tests/CMakeLists.txt:43]: This also needs to be removed.

**Review by ye-luo — COMMENTED** [review]
Please remove SUCCESS_STATUS_CUDA and I will apply cmake-format before merging.

File: tests/CMakeLists.txt (hunk 1 above)
Comment (Hyeondeok-Shin) [reply tests/CMakeLists.txt]: Thanks. SUCCESS_STATUS_CUDA is now completely removed.

**Comment by ye-luo** [comment]
Test this please

**Review by ye-luo — COMMENTED** [review]
Will be last checked by CI.

**Review by ye-luo — APPROVED** [review]

**Pull request merged**